airbyte-cdk/python/airbyte_cdk/sources/declarative/states/dict_state.py

```diff
@@ -0,0 +1,75 @@
+#
+# Copyright (c) 2022 Airbyte, Inc., all rights reserved.
+#
+
+from enum import Enum
+from typing import Mapping, Union
+
+from airbyte_cdk.sources.declarative.interpolation.jinja import JinjaInterpolation
+from airbyte_cdk.sources.declarative.states.state import State
+
+
+def _get_max(*, name, val, other_state):
+    other_val = other_state.get(name)
+    if other_val:
+        return max(val, other_val)
+    else:
+        return val
+
+
+class StateType(Enum):
+    STR = str
+    INT = int
+
+
+class DictState(State):
+    stream_state_field = "stream_state"
+
+    def __init__(self, d: Mapping[str, str] = None, state_type: Union[str, StateType, type] = "STR", config=None):
+        if d is None:
+            d = dict()
```

Comment:
By the look of the tests, it looks like we store templates in `_d`. But why do we need both `context` and this separate mapping field that we then fold into `updated_state`?

Reply:
renamed `_d` to `_templates_to_evaluate`
`_context` contains more information that aren't template strings such as the `stream_state`